test/unit/app/services/asset_manager/attachment_updater_test.rb

```diff
@@ -4,281 +4,234 @@ class AssetManager::AttachmentUpdaterTest < ActiveSupport::TestCase
         AssetManager::AssetUpdater.expects(:call).never
 
-        updater.call(attachment_data, redirect_url: true, draft_status: true)
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, link_header: true)
       end
     end
 
-    describe "access limited updates" do
-      context "when attachment's attachable is access limited" do
-        before do
-          access_limited_object = stub("access-limited-object")
-          @organisation = create(:organisation)
-          AssetManagerAccessLimitation.stubs(:for).with(access_limited_object).returns([@organisation.content_id])
+    context "when the attachment's attachable is a draft" do
+      let(:edition) { create(:draft_news_article) }
+      let(:attachment) { create(:file_attachment, attachable: edition) }
+      let(:assets) do
+        [
+          Asset.new(asset_manager_id: "original_asset_manager_id", variant: Asset.variants[:original], filename: "simple.pdf"),
+          Asset.new(asset_manager_id: "thumbnail_asset_manager_id", variant: Asset.variants[:thumbnail], filename: "thumbnail_simple.pdf.png"),
+        ]
+      end
 
-          attachment_data.stubs(:access_limited?).returns(true)
-          attachment_data.stubs(:access_limited_object).returns(access_limited_object)
-        end
+      before { attachment.attachment_data.assets = assets }
 
-        it "updates the access limited state of all assets" do
-          update_service.expects(:call)
-                          .with(original_asset.asset_manager_id, attachment_data, nil, { "access_limited_organisation_ids" => [@organisation.content_id] })
-          update_service.expects(:call)
-                        .with(thumbnail_asset.asset_manager_id, attachment_data, nil, { "access_limited_organisation_ids" => [@organisation.content_id] })
+      it "marks corresponding assets as draft" do
+        expected_attribute_hash = {
+          "draft" => true,
+        }
 
-          updater.call(attachment_data, access_limited: true)
+        assets.each do |asset|
+          AssetManager::AssetUpdater.expects(:call).with(asset.asset_manager_id, attachment.attachment_data, nil, expected_attribute_hash)
         end
-      end
 
-      context "when attachment's attachable is not access limited" do
-        before do
-          attachment_data.stubs(:access_limited?).returns(false)
-        end
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, draft_status: true)
+      end
 
-        it "updates all assets to have an empty access_limited array" do
-          update_service.expects(:call)
-                          .with(original_asset.asset_manager_id, attachment_data, nil, { "access_limited_organisation_ids" => [] })
-          update_service.expects(:call)
-                        .with(thumbnail_asset.asset_manager_id, attachment_data, nil, { "access_limited_organisation_ids" => [] })
+      it "sets parent_document_url for attachment using draft hostname" do
```

Comment:
As mentioned in our call, it would be nice for this naming to match with the test on line 124 because they're testing the same thing but just draft vs non-draft states